commcare_connect/opportunity/tables.py

```diff
@@ -549,3 +553,285 @@ def date_with_time_popup(table, date):
         date.strftime("%d %b, %Y"),
         date.strftime("%d %b %Y, %I:%M%p"),
     )
+
+
+def header_with_tooltip(label, tooltip_text):
+    return mark_safe(
+        f"""
+        <div class="relative inline-flex justify-center items-center group cursor-default">
+            <span>{label}</span>
+            <i class="fa-regular fa-circle-question text-xs text-slate-400 ml-1 cursor-help"></i>
+            <div class="fixed hidden group-hover:block z-50 pointer-events-none -translate-x-[15%] -translate-y-[70%] transform">
+                <div class="absolute top-full left-1/2 -translate-x-1/2 w-0 h-0 border-l-8 border-r-8 border-t-8 border-l-transparent border-r-transparent border-t-white"></div>
+                <div class="relative bg-white w-28 rounded p-2 text-slate-500 text-xs whitespace-normal break-words">
+                    {tooltip_text}
+                </div>
+            </div>
+        </div>
+    """
+    )
+
+
+class IndexColumn(tables.Column):
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault("verbose_name", "#")
+        kwargs.setdefault("orderable", False)
+        kwargs.setdefault("empty_values", ())
+        super().__init__(*args, **kwargs)
+
+    def render(self, value, record, bound_column, bound_row, **kwargs):
+        table = bound_row._table
+        page = getattr(table, "page", None)
+        if page:
+            start_index = (page.number - 1) * page.paginator.per_page + 1
+        else:
+            start_index = 1
+        if not hasattr(table, "_row_counter") or getattr(table, "_row_counter_start", None) != start_index:
+            table._row_counter = itertools.count(start=start_index)
+            table._row_counter_start = start_index
+        value = next(table._row_counter)
+        return value
+
+
+class BaseOpportunityList(tables.Table):
+    stats_style = "underline underline-offset-2 justify-center"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.use_view_url = False
+
+    index = IndexColumn()
+    opportunity = tables.Column(accessor="name")
+    entity_type = tables.TemplateColumn(
+        verbose_name="",
+        orderable=False,
+        template_code="""
+                <div class="flex justify-start text-sm font-normal text-brand-deep-purple w-fit"
+                     x-data="{
```

Comment:
This could potentially be turned into an alpinejs/tailwind plugin, but this is also the only place it's used.